llvm/test/CodeGen/AIE/aie2/schedule/movxm.mir

```diff
@@ -0,0 +1,413 @@
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $r3 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $r3 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $r3 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            movxm_to_P
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_P
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $p0 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $p0 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $p0 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+
+---
+name:            movxm_to_M
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_M
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $m1 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $m1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $m1 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            movxm_to_DC
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_DC
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $dc1 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dc1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $dc1 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            movxm_to_DJ
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_DJ
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $dj1 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dj1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $dj1 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            movxm_to_DN
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_DN
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $dn1 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $dn1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $dn1 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            movxm_to_S
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: movxm_to_S
+    ; CHECK: BUNDLE implicit-def $r1, implicit-def $s1 {
+    ; CHECK-NEXT:   $r1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $s1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $s1 = MOVXM_lng_cg 0
+    $r1 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_dc_movxm_to_M
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_dc_movxm_to_M
+    ; CHECK: BUNDLE implicit-def $dc1, implicit-def $m1 {
+    ; CHECK-NEXT:   $dc1 = MOVA_lda_cg 0
+    ; CHECK-NEXT:   $m1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: }
+    $m1 = MOVXM_lng_cg 0
+    $dc1 = MOVA_lda_cg 0
+...
+
+---
+name:            mova_to_dc_movxm_to_DC
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: mova_to_dc_movxm_to_DC
+    ; CHECK: $dc1 = MOVXM_lng_cg 0
+    ; CHECK-NEXT: $dc1 = MOVA_lda_cg 0
+    $dc1 = MOVXM_lng_cg 0
+    $dc1 = MOVA_lda_cg 0
```

Comment:
This is output-dependent with a latency of 1. Don't we want to test resource dependence, i.e. use different dc registers?